js/filters/filters-utils.ts

```diff
@@ -0,0 +1,42 @@
+import { Photo } from '../contracts/common';
+import { getRandomIdPack } from '../utils';
+
+const filters = document.querySelector('.img-filters')!;
+const defaultButton = filters.querySelector('#filter-default')!;
+const randomButton = filters.querySelector('#filter-random')!;
+const discussedButton = filters.querySelector('#filter-discussed')!;
+
+const compareComments = ({ comments: commentsA }: Photo, { comments: commentsB }: Photo) => commentsB.length - commentsA.length;
+
+const defaultFilter = (photos: Photo[]) => photos;
+
+const randomFilter = (photos: Photo[]) => {
+	const idPack = getRandomIdPack();
+	const filteredPhotos = photos.filter(({ id }) => idPack.includes(id!));
+
+	return filteredPhotos;
+};
+
+const discussedFilter = (photos: Photo[]) => {
+	const filteredPhotos = photos.slice().sort(compareComments);
+
+	return filteredPhotos;
+};
+
+const filterOptions = {
+	default: defaultFilter,
+	random: randomFilter,
+	discussed: discussedFilter
+};
```

Comment:
Можно добавить сразу тип на фильтры. Тебе же они потом нужны
```suggestion
const filterOptions = {
	default: defaultFilter,
	random: randomFilter,
	discussed: discussedFilter
};

export type FilterOption = keyof typeof filterOptions;
```